Source/WTF/wtf/EnumTraits.h

```diff
@@ -215,38 +215,92 @@ constexpr std::span<const char> enumName()
     return result;
 }
 
-namespace detail {
-
-template<size_t i, size_t end>
-constexpr void forConstexpr(const auto& func)
+template<typename E>
+constexpr std::underlying_type_t<E> enumNamesMin()
 {
-    if constexpr (i < end) {
-        func(std::integral_constant<size_t, i>());
-        forConstexpr<i + 1, end>(func);
+    using Underlying = std::underlying_type_t<E>;
+
+    if constexpr (requires { EnumTraits<E>::min; }) {
+        constexpr Underlying minValue = static_cast<Underlying>(EnumTraits<E>::min);
+        static_assert(std::numeric_limits<Underlying>::min() <= minValue && minValue <= std::numeric_limits<Underlying>::max(),
+            "EnumTraits<E>::min is out of range for the underlying type.");
+        return minValue;
     }
+
+    // Default for both signed and unsigned enums.
+    return 0;
 }
 
+template<typename E>
+constexpr std::underlying_type_t<E> enumNamesMax()
+{
+    using Underlying = std::underlying_type_t<E>;
+
+    if constexpr (requires { EnumTraits<E>::max; }) {
+        constexpr Underlying maxValue = static_cast<Underlying>(EnumTraits<E>::max);
+        static_assert(std::numeric_limits<Underlying>::min() <= maxValue && maxValue <= std::numeric_limits<Underlying>::max(),
```

Comment:
This assert seems like it doesn't do anything because you cast it above? I think you could also maybe cover this by changing the `if` to `if constexpr (requires { { EnumTraits<E>::max } -> std::same_as<Underylying>; })`